packages/components/molecules/f-media-element/src/FlexStyles.js

```diff
@@ -0,0 +1,117 @@
+import { MODIFIER_OPPOSITES_RULES_MAP, MODIFIER_RULES_MAP } from './constants';
+
+class FlexStyles {
+    defaultFlexDirection = '--row';
+
+    defaultFlexReverse = '';
+
+    flexDirection = '--row';
+
+    flexReverse;
+
+    operator;
+
+    operatorOpposite;
+
+    mediaSize;
+
+    constructor ({
+        default: { column: defaultColumn, reverse: defaultReverse },
+        modifier
+    }) {
+        this.defaultColumn = defaultColumn;
+        this.defaultReverse = defaultReverse;
+        this.column = modifier?.column;
+        this.reverse = modifier?.reverse;
+        if (modifier?.rule) {
+            this.rule = modifier?.rule;
+        }
+    }
+
+    /**
+     * Adds the specific part of css class string
+     * @param value
+     * @returns {string}
+     */
+    static getReverseClassSegment (value) {
+        return value ? '--reverse' : '';
+    }
+
+    /**
+     * Adds the specific part of css class string
+     * @param value
+     * @returns {string}
+     */
+    static getDirectionClassSegment (value) {
+        return value ? '--col' : '--row';
+    }
+
+    /**
+     * Function creates a string (css class name) using the previously set class properties
+     * @returns {*}
+     * @private
+     */
+    static createClassName (...values) {
+        if (!values.some(value => value === undefined)) {
+            return values.reduce((a, c) => `${a}${c}`, 'c-mediaElement');
+        }
+        return undefined;
+    }
+
+    /**
+     * Compiles a list of css classes to be applied
+     * @returns {*[]}
+     */
+    get classNames () {
+        return [
+            FlexStyles.createClassName(this.defaultFlexDirection, this.defaultFlexReverse),
+            FlexStyles.createClassName(this.operatorOpposite, this.mediaSize, this.defaultFlexDirection, this.defaultFlexReverse),
+            FlexStyles.createClassName(this.operator, this.mediaSize, this.flexDirection, this.flexReverse)
+        ].filter(value => value !== undefined);
+    }
+
+    /**
+     * Sets the flex direction css string part value
+     * @param value
+     */
+    set column (value) {
```

Comment:
Another dumb class question from me (won't hold up the PR, just I'm interested) – do the set functions get called when the Class is initialised? I couldn't see them getting called anywhere so I'm assuming so (which is why I thought I'd ask!)

Reply:
Yea in this case they are fired when I set the values in the constructor. They can also be set when values are set in a method. They are fired whenever you do `this.column = true` for example, the set for `column` then fires and passes the value that is set in this example `true` you can then use this function to set other variables on the class and do anything with passed value.